src/clp_ffi_js/ir/StructuredIrUnitHandler.cpp

```diff
@@ -0,0 +1,145 @@
+#include "StructuredIrUnitHandler.hpp"
+
+#include <algorithm>
+#include <cctype>
+#include <cstddef>
+#include <iterator>
+#include <memory>
+#include <string>
+#include <string_view>
+#include <type_utils.hpp>
+#include <utility>
+
+#include <clp/ffi/ir_stream/decoding_methods.hpp>
+#include <clp/ffi/KeyValuePairLogEvent.hpp>
+#include <clp/ffi/SchemaTree.hpp>
+#include <clp/ffi/Value.hpp>
+#include <emscripten/val.h>
+#include <spdlog/spdlog.h>
+
+#include <clp_ffi_js/constants.hpp>
+#include <clp_ffi_js/ir/LogEventWithFilterData.hpp>
+
+namespace clp_ffi_js::ir {
+
+namespace {
+/**
+ * Parses the `LogLevel` from an input string.
+ * @param str
+ * @return
+ */
+auto parse_log_level(std::string_view str) -> LogLevel;
+
+auto parse_log_level(std::string_view str) -> LogLevel {
```

Comment:
We'll hear from @kirkrodrigues whether we want need forward declarations for helpers like this.